include/arch/arm/cortex_m/scripts/linker.ld

```diff
@@ -299,11 +299,13 @@ SECTIONS
 		_image_ram_start = .;
 		_app_smem_start = .;
 		APP_SMEM_SECTION()
-		_app_smem_end = .;
-		. = ALIGN(4);
+		/* 32-byte align end of app shared area to make MPU friendly */
+		. = ALIGN(_app_data_align);
+
 	} GROUP_DATA_LINK_IN(RAMABLE_REGION, ROMABLE_REGION)
 #endif  /* CONFIG_MPU_REQUIRES_POWER_OF_TWO_ALIGNMENT */
 
+	_app_smem_end = .;
```

Comment:
we need the end of smem before there is any padding.
This will change the memory alignment requirement for power of 2 MPUs.
Can you please move this to line 302.

Reply:
@AdithyaBaglody woulnd't it actually be more accurate, if app_smem_end is after line 303, so it is already 32-byte-aligned? 
Then, basically app_smem_end = app_ram_start, which is exactly what we want.

I see the point of putting app_smem_end inside the #else #endif CONFIG_MPU_REQUIRES_POWER_OF_TWO_ALIGNMENT .

Actually, this was my primary intention, i.e. fix the bug for ARMv8-M which does _not_ require power-of-two alignment.

Reply:
@AdithyaBaglody yes, so I added my modification in the #else only, that is, for MPUs that do _not_ have the requirement for power-of-two alignment.

Please, see the latest version. So, effectively, the padding will apply to MPUs without the requirement for power-of-two. This is meant for ARMv8-M unmodified MPU. It may affect the NXP_MPU, too, but there is no harm, really, to add some extra padding, in this case, IMO. The _app_smem_end will, anyways contain the end of the section.

Reply:
>yes, so I added my modification in the #else only, that is, for MPUs that do not have the requirement for power-of-two alignment.

oh my bad i thought it was under power of 2. Looked at the #endif and thought it was under it.